docs/examples/Rocket_Car.rst

```diff
@@ -0,0 +1,875 @@
+
+    B2_cons = [me.dot(B2_ground.vel(N).simplify(), uv) for uv in A1]
+    sm.trigsimp(B2_cons)
+
+
+.. jupyter-execute::
+
+    eq3 = sm.Eq(B2_cons[0].simplify().subs(u), 0)
+    eq3
+
+
+.. jupyter-execute::
+
+    eq4 = sm.Eq(B2_cons[1].simplify().subs(u), 0)
+    eq4
+
+
+Nonholonomic Constraints: :math:`B_3`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. jupyter-execute::
+
+    B3_center = P.locatenew('B_3_center', b * A2.y)
+    B3_center.pos_from(P)
+
+
+.. jupyter-execute::
+
+    B3_center.v2pt_theory(P, N, A2)
+    B3_center.vel(N).express(A2).simplify()
+
+
+.. jupyter-execute::
+
+    B3_ground = B3_center.locatenew('B_3_ground', r2 * -A2.z)
+    B3_ground.pos_from(B3_center)
+
+
+.. jupyter-execute::
+
+    B3_ground.v2pt_theory(B3_center, N, B3)
+    B3_ground.vel(N).simplify()
+
+
+.. jupyter-execute::
+
+    B3_cons = [me.dot(B3_ground.vel(N).simplify(), uv) for uv in A2]
+    sm.trigsimp(B3_cons)
+
+
+
+.. jupyter-execute::
+
+    eq5 = sm.Eq(B3_cons[0].simplify().subs(u), 0)
+    eq5
+
+
+.. jupyter-execute::
+
+    eq6 = sm.Eq(B3_cons[1].simplify().subs(u), 0)
+    eq6
+
+
+
+Nonholonomic Constraints: :math:`B_4`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. jupyter-execute::
+
+    B4_center = P.locatenew('B_4_center', b * -A2.y)
+    B4_center.pos_from(P)
+
+
+.. jupyter-execute::
+
+    B4_center.v2pt_theory(P, N, A2)
+    # B4_center.vel(N).express(A2).simplify() # Invalid JSON
+
+
+
+.. jupyter-execute::
+
+    B4_ground = B4_center.locatenew('B_4_ground', r2 * -A2.z)
+    B4_ground.pos_from(B4_center)
+
+
+
+.. jupyter-execute::
+
+    B4_ground.v2pt_theory(B4_center, N, B4)
+    B4_ground.vel(N).simplify()
+
+
+.. jupyter-execute::
+
+    B4_cons = [me.dot(B4_ground.vel(N).simplify(), uv) for uv in A2]
+    sm.trigsimp(B4_cons)
+
+
+.. jupyter-execute::
+
+    eq7 = sm.Eq(B4_cons[0].simplify().subs(u), 0)
+    eq7
+
+
+.. jupyter-execute::
+
+    eq8 = sm.Eq(B4_cons[1].simplify().subs(u), 0)
+    eq8
+
+
+:math:`\text{LHS} \Longleftrightarrow \text{RHS}\ \text{in}\ z_1, z_2 \rightarrow \text{Equations}\ 9, 10`
+
+.. jupyter-execute::
+
+    eq9 = sm.Eq(A1.ang_vel_in(A2).dot(A1.z), u1)
+    eq9
+
+
+.. jupyter-execute::
+
+    eq10 = sm.Eq(Q.vel(N).dot(A1.x), u2)
+    eq10
+
+
+Solving the System of Linear Equations
+--------------------------------------
+
+The system of equations is linear in :math:`\dot{q}_1, \dot{q}_2,...`
+
+.. jupyter-execute::
+
+    solution = sm.solve([eq1, eq2, eq3, eq4, eq5, eq6, eq7, eq8, eq9, eq10], q1.diff(), q2.diff(), q3.diff(),  q4.diff(), q5.diff(), q6.diff(), q7.diff(), q8.diff())
+
+.. jupyter-execute::
+
+    solution
+
+
+
+.. jupyter-execute::
+
+    sollist_keys = list(solution.keys())
+    sollist_keys
+
+
+
+.. jupyter-execute::
+
+    sollist_values = list(solution.values())
+    sollist_values
+
+
+
+.. jupyter-execute::
+
+    sollist_values_simple = []
+    for i in range(8):
+       sollist_values_simple.append(sm.factor(TR2(sollist_values[i]).simplify()))
```

Comment:
This line is giving an error:

```
Extension error:

Cell raised uncaught exception:

---------------------------------------------------------------------------

IndexError                                Traceback (most recent call last)

<ipython-input-49-e76e181d41dd> in <module>()

      1 sollist_values_simple = []

      2 for i in range(8):

----> 3    sollist_values_simple.append(sm.factor(TR2(sollist_values[i]).simplify()))

IndexError: list index out of range
```

Maybe change `8` to `len(something)` so you have the correct range?